### DISCUSSION TITLE
Is it possible to register a concrete instance, so it is used instead of a Mock by AutoMoq?

### DISCUSSION THREAD

**Discussion by voroninp** [AutoFixture/AutoFixture#1288] [Q&A] [answered]
I need to inject `TestableEndpointInstance` as `IMessageSession` dependency to SUT.

After I freeze it:
```CSharp
session = new TestableEndpointInstance();
Fixture.Freeze<IMessageSession>(session);
```
I see in debug that `AutoMoq` customization still generates `Mock<IMessageSession>`.
Can I make Autofixture use registered dependencies instead of mocks or do I have switch back to manual SUT creation?

**Comment by aivascu** [comment, marked as answer]
@voroninp you can use either `Inject<T>()` or `Register<T>()` to instruct AutoFixture to resolve a specific instance for an interface.
https://autofixture.io/api/AutoFixture.FixtureRegistrar.html#methods

**Reply by voroninp** [reply]
My bad, I just incorrectly used  `Freeze<T>(T seed)`
